packages/components/src/public/QueryModel/GridPanel.tsx

```diff
@@ -501,11 +512,11 @@ export class GridPanel<T = {}> extends PureComponent<Props<T>, State> {
         }
     };
 
-    selectRow = (row, event): void => {
+    selectRow = (row: Map<string, any>, event: ChangeEvent<HTMLInputElement>): void => {
         const { model, actions } = this.props;
         const checked = event.target.checked === true;
-        // Have to call toJS() on the row because <Grid /> converts rows to Immutable objects.
-        actions.selectRow(model.id, checked, row.toJS());
+        // Look through to the nativeEvent to determine if the shift key is engaged.
+        actions.selectRow(model.id, checked, row.toJS(), (event.nativeEvent as any).shiftKey ?? false);
```

Comment:
I've seen this `?? false` a couple times now to default this param. Could that default just be done in withQueryModels.tsx via:
`selectRow: (id: string, checked: boolean, row: { [key: string]: any }, useSelectionPivot = false) => void;`